core-drawer-panel.html

```diff
@@ -75,344 +75,425 @@
+            -this.edgeSwipeSensitivity : this.edgeSwipeSensitivity));
+        this.peeking = true;
+      },
+
+      stopEdgePeek: function() {
         if (this.peeking) {
-          if (Math.abs(e.dx) <= this.edgeSwipeSensitivity) {
-            return; // Ignore trackx until we move past the edge peek.
-          }
           this.peeking = false;
+          this.moveDrawer(null);
         }
-        this.moveDrawer(this.translateXForDeltaX(e.dx));
-      }
-    },
+      },
 
-    trackEnd : function(e) {
-      if (this.dragging) {
-        this.dragging = false;
-        this.transition = true;
-        this.moveDrawer(null);
+      downHandler: function(e) {
+        if (!this.dragging && this.isMainSelected() && this.isEdgeTouch(e)) {
+          this.startEdgePeek();
+        }
+      },
+
+      upHandler: function(e) {
+        this.stopEdgePeek();
+      },
+
+      onClick: function(e) {
+        if (e.target && this.toggleAttribute &&
+            e.target.hasAttribute(this.toggleAttribute)) {
+          this.togglePanel();
+        }
+      },
+
+      isEdgeTouch: function(e) {
+        return !this.disableEdgeSwipe && this.swipeAllowed() &&
+          (this.rightDrawer ?
+            e.pageX >= this.offsetWidth - this.edgeSwipeSensitivity :
+            e.pageX <= this.edgeSwipeSensitivity);
+      },
 
+      trackStart : function(e) {
+        if (this.swipeAllowed()) {
+          this.dragging = true;
+
+          if (this.isMainSelected()) {
+            this.dragging = this.peeking || this.isEdgeTouch(e);
+          }
+
+          if (this.dragging) {
+            this.width = this.$.drawer.offsetWidth;
+            this.transition = false;
+            e.preventTap();
+          }
+        }
+      },
+
+      translateXForDeltaX: function(deltaX) {
+        var isMain = this.isMainSelected();
         if (this.rightDrawer) {
-          this.selected = e.xDirection > 0 ? 'main' : 'drawer';
+          return Math.max(0, isMain ? this.width + deltaX : deltaX);
         } else {
-          this.selected = e.xDirection > 0 ? 'drawer' : 'main';
+          return Math.min(0, isMain ? deltaX - this.width : deltaX);
         }
-      }
-    },
+      },
 
-    transformForTranslateX: function(translateX) {
-      if (translateX === null) {
-        return '';
-      }
-      return this.hasWillChange ? 'translateX(' + translateX + 'px)' : 
-          'translate3d(' + translateX + 'px, 0, 0)';
-    },
+      trackx : function(e) {
+        if (this.dragging) {
+          if (this.peeking) {
+            if (Math.abs(e.dx) <= this.edgeSwipeSensitivity) {
+              return; // Ignore trackx until we move past the edge peek.
+            }
+            this.peeking = false;
+          }
+          this.moveDrawer(this.translateXForDeltaX(e.dx));
+        }
+      },
+
+      trackEnd : function(e) {
+        if (this.dragging) {
+          this.dragging = false;
+          this.transition = true;
+          this.moveDrawer(null);
```

Comment:
I'm not totally sure what to expect from passing `null` here. I had to trace through the call to get it. Not sure what to do about that, but felt it was worth mentioning.